src/data/error.js

```diff
@@ -10,7 +12,29 @@ export const error = writable(defaultError)
 init()
 
 function init() {
+  // the incoming object here is from the browser https://developer.mozilla.org/en-US/docs/Web/API/PromiseRejectionEvent
   window.onunhandledrejection = ({reason}) => {
+    // `reason` will contain whatever was thrown (which should only be from `throwError` below)
+
+    if (reason.code === 401) {
+      // we need to get the user over to the login page but we may want to set up some additional
+      // handling in case they were in the middle of something in the app already, i.e., they
+      // already authenticated but their credenitals expired while they were looking at requests.
+      const currentRoute = get(location)
+      let loginRoute = '/login'
+      reason.message = ''  // we'll only show the user info if it's helpful
+
+      if (! ['/', loginRoute].includes(currentRoute)) {
+        // they were on some page in the app already and lost their credentials,
+        // we'll want to give them a little info and get them back to where they were.
+        loginRoute += `?return-to=${currentRoute}`
```

Comment:
We might need to url-encode the `currentRoute`, since it's being used as the value for a query-string parameter (and thus, I believe, is expected to be url-encoded).

Reply:
help me understand this one better...do those concerns change since it's a SPA and we're using the library to push?

Reply:
e.g., if it needed to be encoded, wouldn't it already be? or are you saying since we're introducing the `?`, `-` and `=`, we need to do it?

Reply:
Your last comment is it: since we are assembling the query string by inserting the question mark, parameter name, equals sign, and parameter value, we're the ones responsible for url-encoding the parameter value.

Most of the time we'll just sort of "get lucky" because a url-encoded string will match the normal string itself, but occasionally there will be a difference, and the "right" way to provide a query-string parameter's value is to url-encode it.

Reply:
ok, we'll need to review other places then too and just look at the url-encoding across the app...there might be other places we'd want to do it.